### PR TITLE
ci: skips preview publishing if version already exists

### PR DESCRIPTION
**Pull request by kleinfreund**
# Summary

Skips trying to publish a preview package if a package with the same version already exists. This is necessary in order to be able to re-run flaky tests.

## Notes

Someone with better Shell foo should be looking over this

## PR Checklist

* [x] Does not introduce dependencies
* [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [x] **Tests pass:** check the output of yarn test
* [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [x] **Framework style:** abides by the essential rules in Vue's style guide
* [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -71,11 +71,24 @@ jobs:
...
+            | grep version \
+            | head -1 \
+            | awk -F: '{ print $2 }' \
+            | sed 's/[", ]//g')
+          package=@kong/kongponents@"${package_version}"
+
+          npm show "${package}" >/dev/null 2>&1 && npm_show_status=0 || npm_show_status=1
+          if [ $npm_show_status -eq 0 ]; then
+            echo "Package ${package} is already published. Skipping publishing."
+            exit 0
```
Comment (adamdehaven) [inline .github/workflows/test.yml:86]: We don't want to exit here without returning success as this will also prevent the tests from running.

Comment (kleinfreund) [reply .github/workflows/test.yml:86]: Are you sure? Shouldn’t this just exit this step with a successful status?

Comment (adamdehaven) [reply .github/workflows/test.yml:86]: Yep. 0. Ignore this 🤦🏼‍♂️

**Review by ValeryG — CHANGES_REQUESTED** [review]

**Review by ValeryG — APPROVED** [review]

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**